src/ILogger/ApplicationInsightsLoggingBuilderExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ApplicationInsightsLoggingBuilderExtensions.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2013
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Extensions.Logging
+{
+    using System;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.DependencyInjection.Extensions;
+    using Microsoft.Extensions.Logging.ApplicationInsights;
+
+    /// <summary>
+    /// Extensions methods to add and configure application insights logger.
+    /// </summary>
+    public static class ApplicationInsightsLoggingBuilderExtensions
```

Comment:
The instrumentation key should be a parameter here’s the caller shouldn’t have to drop down to service collection shenanigans to configure it.

Reply:
Pulling Instrumentation key alone is not enough. TelemetrySinks, TelemetryChannels, TelemetryInitializers and TelemetryProcessors would still need to be setup. However these are not always required to be setup.

One option is to have an extension method which takes instrumentation key to get people up and running while rest of the stuff can be added later using .Configure<TelemetryConfiguration> if needed.